libnmstate/nm/route.py

```diff
@@ -0,0 +1,147 @@
+                auto_origin_value,
+                table_id,
+                iface_name)
+            if route_entry:
+                ret.append(route_entry)
+    ret.sort(key=itemgetter(LinuxRoute.TABLE_ID, LinuxRoute.DESTINATION))
+    return ret
+
+
+def _should_skip(dst, skips):
+    '''
+    Check whether specified route destination is in skip networks.
+    '''
+    for network in skips:
+        if iplib.is_subnet_of(dst, network):
+            return True
+    return False
+
+
+def _get_route_origin(active_connection, static_routes, dst,
+                      is_auto_route_func, auto_origin_value):
+    if dst not in static_routes and is_auto_route_func(active_connection):
+        return auto_origin_value
+    return LinuxRouteOrigin.STATIC
+
+
+def _get_route_table_id(active_connection, ip_profile):
+    table_id = ip_profile.props.route_table if ip_profile \
+        else iplib.MAIN_ROUTE_TABLE_ID
+    if table_id == iplib.MAIN_ROUTE_TABLE_ID:
+        return iplib.get_route_table_id(iplib.MAIN_ROUTE_TABLE_NAME)
+    return table_id
+
+
+def _get_iface_name(active_connection):
+    """
+    Return interface name for active_connection, return None if error.
+    """
+    # If we have master device, we use it, else we use first device.
+    if hasattr(active_connection, 'master'):
+        return active_connection.master.get_iface()
+    else:
+        devs = active_connection.get_devices()
+        if devs:
+            return devs[0].get_iface()
+    return None
+
+
+def _get_static_routes(active_connection, ip_profile):
+    static_routes = []
+    if ip_profile:
+        for route in ip_profile.props.routes:
+            static_routes.append(
+                '{ip}/{prefix}'.format(
+                    ip=route.get_dest(), prefix=route.get_prefix()))
+    return static_routes
+
+
+def _libnm_route_to_route(libnm_route, active_connection, skip_networks,
+                          static_routes, is_auto_route_func, auto_origin_value,
+                          table_id, iface_name):
+    dst = '{ip}/{prefix}'.format(
+        ip=libnm_route.get_dest(), prefix=libnm_route.get_prefix())
+    if _should_skip(dst, skip_networks):
+        return None
+    route_origin = _get_route_origin(
+        active_connection,
+        static_routes,
+        dst,
+        is_auto_route_func,
+        auto_origin_value)
+    next_hop = libnm_route.get_next_hop()
+    if not next_hop:
+        next_hop = ''
+    metric = int(libnm_route.get_metric())
+
+    return {
+        LinuxRoute.ORIGIN: route_origin,
+        LinuxRoute.TABLE_ID: table_id,
+        LinuxRoute.TABLE_NAME: iplib.get_route_table_name(table_id),
+        LinuxRoute.DESTINATION: dst,
+        LinuxRoute.NEXT_HOP_INTERFACE: iface_name,
+        LinuxRoute.NEXT_HOP_ADDRESS: next_hop,
+        LinuxRoute.METRIC: metric,
+    }
+
+
+def _get_skip_networks(ip_cfg):
+    """
+    Currently libnm cannot manage the direct route of interface's LAN:
```

Comment:
... cannot manage directed connected routes.
Example:
   eth1 with an IP address ...
   has the following direct route: ...

But I do not understand why you need to hide it.

Reply:
As libnm cannot edit on direct route, showing them seems like a noise to me.